### PR TITLE
[KOA-4652] Migrate BPKChip to new icon api

### PR DESCRIPTION
**Pull request by k0nserv**
Remember to include the following changes:

+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

File: Example/Backpack/BPKAppDelegate.m [hunk 1]
```diff
@@ -39,10 +39,15 @@ @implementation BPKAppDelegate
...
-        [BPKFont setFontDefinition:relativeFontDefinition];
+
+    BOOL useRelative = ![NSProcessInfo.processInfo.arguments containsObject:@"DISABLE_RELATIVE"];
+
+    if (useRelative) {
+        id<BPKFontDefinitionProtocol> relativeFontDefinition = [BPKRelativeFontDefinition new];
+        UIFont *relativeTestFont = [UIFont fontWithName:relativeFontDefinition.regularFontFace size:12];
+        if (relativeTestFont != nil) {
+            [BPKFont setFontDefinition:relativeFontDefinition];
+        }
```
Comment (georgegillams) [inline Example/Backpack/BPKAppDelegate.m:50]: Would it be cleaner if we did something like `useRelative = !env["DISABLE_RELATIVE"] && relativeAvailable`

Then underneath have 
```
if (useRelative) {
             [BPKFont setFontDefinition:relativeFontDefinition];
         }
         ```

File: UNRELEASED.md [hunk 2]
```diff
@@ -9,6 +9,8 @@
   - In Objective-C some options are no longer supported e.g. the ability to show a notification dot and the support for badges.
   - The component adopts the new `BPKIcon` API and enforces the correct usage of icons with the appropriate horizontal navigation size at compile time.
   - Due to a bug in Swift/UIKit with generic views the component no longer supports theming. Theming can still be achieved with an explicit wrapping view that integrates with `UIAppearance`.
+- Backpack/Chip:
+  - The type of `iconName` has changed from the deprecated `BPKIconName` to `BPKIconNameLarge`. Some icons previously used might to be available.
```
Comment (georgegillams) [inline UNRELEASED.md:13]: This shouldn't be the case as using them would have resulted in a runtime error, but I guess it does no harm to mention

Comment (k0nserv) [reply UNRELEASED.md:13]: Right, probably still worth mentioning?

File: Example/Backpack/BPKAppDelegate.m (hunk 1 above)
Comment (k0nserv) [reply Example/Backpack/BPKAppDelegate.m]: Like so https://github.com/Skyscanner/backpack-ios/pull/1002/commits/357f3c398d85a44a3d245732ffd6595b438ce3d3?

File: Example/Backpack/BPKAppDelegate.m [hunk 3]
```diff
@@ -41,7 +41,10 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
     id<BPKFontDefinitionProtocol> relativeFontDefinition = [BPKRelativeFontDefinition new];
     UIFont *relativeTestFont = [UIFont fontWithName:relativeFontDefinition.regularFontFace size:12];
-    if (relativeTestFont != nil) {
+    BOOL relativeAvailable = relativeTestFont != nil;
+    BOOL useRelative = ![NSProcessInfo.processInfo.arguments containsObject:@"DISABLE_RELATIVE"] && relativeAvailable;
+
+    if (useRelative) {
```
Comment (georgegillams) [inline Example/Backpack/BPKAppDelegate.m:47]: Noice

**Review by georgegillams — APPROVED** [review]

**Comment by k0nserv** [comment]
Dunno why the Snyk steps are broken, merging anyway

**Pull request merged**